packages/sdk-components-react-radix/src/radix-collapsible.tsx

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable react/display-name */
+// We can't use .displayName until this is merged https://github.com/styleguidist/react-docgen-typescript/pull/449
+
+import {
+  type ReactNode,
+  type ForwardRefExoticComponent,
+  type ComponentPropsWithRef,
+  forwardRef,
+  Children,
+} from "react";
+import {
+  Collapsible,
+  CollapsibleTrigger,
+  CollapsibleContent,
+} from "@radix-ui/react-collapsible";
+import {
+  type WebstudioAttributes,
+  splitPropsWithWebstudioAttributes,
+} from "@webstudio-is/react-sdk";
+
+export const RadixCollapsible: ForwardRefExoticComponent<
+  Omit<ComponentPropsWithRef<typeof Collapsible>, "defaultOpen" | "asChild">
+> = Collapsible;
+
+/**
+ * CollapsibleTrigger is HTML-less components.
+ * To make them work in our system, we wrap their attributes with a div that has a display: contents property.
+ *
+ * These divs function like fragments, with all web studio-related attributes attached to them.
+ */
+const displayContentsStyle = { display: "contents" };
+
+/**
+ * We're not exposing the 'asChild' property for the Trigger.
+ * Instead, we're enforcing 'asChild=true' for the Trigger and making it style-less.
+ * This avoids situations where the Trigger inadvertently passes all styles to its child,
+ * which would prevent us from displaying styles properly in the builder.
+ */
+export const RadixCollapsibleTrigger = forwardRef<
+  HTMLDivElement,
+  WebstudioAttributes & { children: ReactNode }
+>(({ children, ...props }, ref) => {
+  const firstChild = Children.toArray(children)[0];
+  const [webstudioAttributes, restProps] =
+    splitPropsWithWebstudioAttributes(props);
+  return (
+    <div ref={ref} style={displayContentsStyle} {...webstudioAttributes}>
```

Comment:
shouldn't this div be part of builder instead of SDK? We could have a meta that says something like "tagless" in which case we wrap it into that div with display contents.

reason for this is to make it simpler providing components to builder, since this is purely builder specific thing, a published site doesn't need this

Reply:
The issue that it's not builder related only. We use components + component wrapper on raw Canvas too. 
(And to support refs we forward refs on `WebstudioComponent` too)

So logic need to be moved on Canvas too. To avoid various `refs` issues. (Like ref can't be passed to functional component)

Would need aditional meta info, and will cause additional questions like:

Why we forward ref for this component and do not forward ref for that component. (answer because it will be wrapped somewhere else with div fragment)

Why this component receives webstudio id property and that component do not receive webstudio id property
(we can't assign webstudioId to components like triggers) (answer - because we will add div somewhere else with webstudio id)

Also rendering tree at the builder and on canvas will be different, so easy to lost something too, even its display: contents it will have some side-effects. (Events, etc)

Even now it's hard to think about this types of components, and then everything will be implcit I feel it will be harder.

Reply:
Also as of now I made `asChild` on triggers to be always `true`. That makes trigger `element less` component. But with `asChild=false` it become standalone element. 
So slots are dynamic by they nature.
If some day we will find that we need to support `asChild` - how we will pass that information through the meta. 

(I will need to remove asChild from props of other components for now as they will cause bug)

Reply:
1. we wrap sdk component anyways implicitely, following your logic we should be wrapping explicitly too
2. having inline style around every wrapped component isn't ideal
3. ist still not clear what needs to be passed to that wrapper component and why. If I had to create my own SDK component, I wouldn't know what to do in this case either

It's a system behavior leaked into user sdk.

Reply:
Btw even though the wrapper here is made explicit, the need for it or how it works is implicit. It will require understanding a lot of stuff and a lot of comments.

Reply:
Also this will need to happen in every such sdk component vs. single place inside the WebstudioComponent

Reply:
More so if we need to change this thing in any way, this will need to happen in every such component. Having it centrally will make it fixable in one place.

Reply:
I will wait a little bit until all components will be +- stable. Will work on Canvas (not tested) have no issues etc. Im not even near sure that we have covered all cases now.
Also as I wrote above it’s not meta information, as because of refs will need knowledge at Canvas level too

Reply:
Probably must be solved other way. For example supporting in the builder html less components without wrapper. (like htmlless component need to select all children on selection etc)

Reply:
Yeah, we will definitely need to support domless components soon anyways.